### PR TITLE
renderers: test empty values

### PR DESCRIPTION
**Pull request by pared**
Related: iterative/vscode-dvc#1520

### CODE REVIEW

File: tests/test_vega.py [hunk 1]
```diff
@@ -9,6 +9,16 @@
 # pylint: disable=missing-function-docstring
```
Comment (daavoo) [inline tests/test_vega.py:9]:
```suggestion
# pylint: disable=missing-function-docstring, use-implicit-booleaness-not-comparison
```

File: tests/test_vega.py [hunk 2]
```diff
@@ -6,7 +6,17 @@
 from dvc_render.vega import BadTemplateError, VegaRenderer
 from dvc_render.vega_templates import NoFieldInDataError, Template
 
-# pylint: disable=missing-function-docstring
+# pylint: disable=missing-function-docstring, use-implicit-booleaness-not-comparison
```
Comment (daavoo) [inline tests/test_vega.py:9]:
```suggestion
# pylint: disable=missing-function-docstring, C1803
```

File: src/dvc_render/base.py [hunk 3]
```diff
@@ -18,10 +18,10 @@ class Renderer(abc.ABC):
 
     EXTENSIONS: Iterable[str] = {}
 
-    def __init__(self, datapoints: Dict, name: str, **properties):
-        self.datapoints = datapoints
-        self.name = name
-        self.properties = properties
+    def __init__(self, datapoints: List, name: str, **properties):
```
Comment (skshetry) [inline src/dvc_render/base.py:21]:
```suggestion
    def __init__(self, datapoints: List = None, name: str = None, **properties):
```

As you are already expecting a `NoneType`, this should make it more obvious.

File: src/dvc_render/base.py [hunk 4]
```diff
@@ -18,10 +18,10 @@ class Renderer(abc.ABC):
...
+        self.datapoints = datapoints or []
+        self.name = name or ""
+        self.properties = properties or {}
```
Comment (skshetry) [inline src/dvc_render/base.py:24]: If there are no kwargs passed, it's an empty dict by default.
```suggestion
        self.properties = properties
```

**Review by skshetry — APPROVED** [review]

File: tests/test_vega.py [hunk 5]
```diff
@@ -6,7 +6,17 @@
...
+
+
+def test_init_empty():
+    renderer = VegaRenderer(None, None)
+
+    assert renderer.datapoints == []
+    assert renderer.name == ""
+    assert renderer.properties == {}
+
+    renderer.partial_html()
```
Comment (daavoo) [inline tests/test_vega.py:19]: Should we check the contents of the resulting template?

Comment (pared) [reply tests/test_vega.py]: done

**Review by daavoo — APPROVED** [review]

**Pull request merged**